File: gradle-module-plugin/src/functionalTest/resources/gateway/GatewayHook.java
package <PACKAGE_PATH>;

import java.util.List;

import com.inductiveautomation.ignition.common.expressions.ExpressionFunctionManager;
import com.inductiveautomation.ignition.common.licensing.LicenseState;
import com.inductiveautomation.ignition.common.script.ScriptManager;
import com.inductiveautomation.ignition.common.xmlserialization.deserialization.XMLDeserializer;
import com.inductiveautomation.ignition.common.xmlserialization.serialization.XMLSerializer;
import com.inductiveautomation.ignition.gateway.AbstractGatewayModuleHook;
import com.inductiveautomation.ignition.gateway.clientcomm.ClientReqSession;
import com.inductiveautomation.ignition.gateway.web.models.INamedTab;

class <MODULE_NAME>GatewayHook extends AbstractGatewayModuleHook {
    @Override
    public Object getRPCHandler(ClientReqSession session, String projectName) {
        return null;
    }

    @Override
    public List<? extends IHomepagePanelDescriptor> getHomepagePanels() {
        return null;
    }

    @Override
    public List<? extends INamedTab> getStatusPanels() {
        return null;
    }

    @Override
    public void configureDeserializer(XMLDeserializer deserializer) {

    }

    @Override
    public void configureSerializer(XMLSerializer serializer) {

    }

    @Override
    public void configureFunctionFactory(ExpressionFunctionManager factory) {

    }

    @Override
    public void notifyLicenseStateChanged(LicenseState licenseState) {

    }

    @Override
    public void initializeScriptManager(ScriptManager manager) {

    }
}

File: module-generator/core/src/main/resources/templates/hook/GatewayHook.kt
package <PACKAGE_ROOT>.gateway

import com.inductiveautomation.ignition.common.expressions.ExpressionFunctionManager
import com.inductiveautomation.ignition.common.licensing.LicenseState
import com.inductiveautomation.ignition.common.script.ScriptManager
import com.inductiveautomation.ignition.common.xmlserialization.deserialization.XMLDeserializer
import com.inductiveautomation.ignition.common.xmlserialization.serialization.XMLSerializer
import com.inductiveautomation.ignition.gateway.AbstractGatewayModuleHook
import com.inductiveautomation.ignition.gateway.clientcomm.ClientReqSession
import com.inductiveautomation.ignition.gateway.web.models.INamedTab

/**
 * Gateway scoped hook implementation
 */
class <MODULE_CLASSNAME>GatewayHook: AbstractGatewayModuleHook() {


    fun getRPCHandler(session: ClientReqSession?, projectName: String?): Any? {
        return null
    }

    val homepagePanels: List<Any?>?
        get() = null

    val statusPanels: List<Any?>?
        get() = null

    fun configureDeserializer(deserializer: XMLDeserializer?) {

    }

    fun configureSerializer(serializer: XMLSerializer?) {

    }
    fun configureFunctionFactory(factory: ExpressionFunctionManager?) {

    }

    fun notifyLicenseStateChanged(licenseState: LicenseState?) {

    }

    fun initializeScriptManager(manager: ScriptManager?) {

    }
}
